src/util/ser.rs

```diff
@@ -0,0 +1,373 @@
+use std::result::Result;
+use std::io::{self, Read, Write};
+use std::fmt;
+use std::collections::HashMap;
+use std::hash::Hash;
+use std::mem;
+
+use secp256k1::Secp256k1;
+use secp256k1::key::PublicKey;
+use bitcoin::blockdata::script::Script;
+use bitcoin::util::hash::Sha256dHash;
+use std::marker::Sized;
+
+pub const MAX_VEC_SIZE: usize = 32 * 1024 * 1024;
+
+pub enum Error {
+	CorruptedData,
+	BadPublicKey,
+	IoError(io::Error),
```

Comment:
Why not simply it's named `Io` like most crates do?

Reply:
Replaced to `Io` now.